base/threads.jl

```diff
@@ -11,25 +11,4 @@ include("threadingconstructs.jl")
 include("atomics.jl")
 include("locks-mt.jl")
 
-
-"""
-    resize_nthreads!(A, copyvalue=A[1])
-
-Resize the array `A` to length [`nthreads()`](@ref).   Any new
-elements that are allocated are initialized to `deepcopy(copyvalue)`,
-where `copyvalue` defaults to `A[1]`.
-
-This is typically used to allocate per-thread variables, and
-should be called in `__init__` if `A` is a global constant.
-"""
```

Comment:
I think a lot of packages use this

Reply:
Yeah, I think we should keep it and maybe not even deprecate it yet. I think it's a good signaling channel just in case we break the "static `nthreads`" restriction in the future. Or a much better direction is to provide a declarative "thread-local storage" API (as opposed to task-local) that can be implemented now and in the future, and _then_ provide the deprecation message for a clear upgrade path.

(Of course, the thread-local property cannot be used in generic Julia code using the proper surface semantics of Julia language. But I'm hoping that we can get there.)

Reply:
I am confused guys. We approved deleting this months ago (https://github.com/JuliaLang/julia/pull/42302), so I included it here so that that would occur early in the current release cycle, and unblock other work (e.g. #42302). Of course we won't backport this, to give people enough time to switch to task-local-storage. But they should have been using TLS before anyways.

Reply:
> We approved deleting this months ago (#42302)

I didn't know this. FWIW, I'm still arguing strongly that `nthreads` should be the total number of worker threads in all pools  https://github.com/JuliaLang/julia/pull/42302#discussion_r784380643

Reply:
...and @jpsamaroo seems to be OK with `nthreads` being the total number of threads https://github.com/JuliaLang/julia/pull/42302#issuecomment-1049104812

Reply:
@tkf I agree with that interpretation, but I also hope to entirely delete the `nthreads` function next too, so that people stop trying to rely upon it being a defined constant (you can still call `Base.JLOptions().nthreads` if you want to know the initial user-configured size of the default worker pool)